ext/prism/extension.c

```diff
@@ -599,8 +599,7 @@ parse_lex_input(pm_string_t *input, const pm_options_t *options, bool return_nod
 
     VALUE source_string = rb_str_new((const char *) pm_string_source(input), pm_string_length(input));
     VALUE offsets = rb_ary_new();
-    VALUE source_argv[] = { source_string, LONG2NUM(parser.start_line), offsets };
-    VALUE source = rb_class_new_instance(3, source_argv, rb_cPrismSource);
+    VALUE source = rb_funcall(rb_cPrismSource, rb_intern("for"), 3, source_string, LONG2NUM(parser.start_line), offsets);
```

Comment:
Last change, and then this looks good.

I don't want to `rb_intern("for")` every time this method is called, because it's called whenever we parse. Instead can you eagerly get the ID when prism is initialized? There's prior art here with stuff like `rb_option_id_command_line`.

Reply:
Done! We're still calling `rb_intern` in `api_node.c.erb`, not sure if you want to get rid of that one too.